hs_core/tests/api/native/test_copy_resource.py

```diff
@@ -448,3 +440,9 @@ def test_copy_composite_resource(self):
             self.composite_resource.delete()
         if new_composite_resource:
             new_composite_resource.delete()
+
+
+def _get_relation_meta_derived_from(resource):
+    today = date.today().strftime("%m/%d/%Y")
+    derived_from = "{}, accessed on: {}".format(resource.get_citation(), today)
+    return derived_from
```

Comment:
This is kind of related to my question about regarding using citation as the value of the relation. If accessed on date is part of the metadata value of derived from, it means every time when a resource is accessed, the relation value needs to be updated. This may explain why you call that update metadata function is page processor method. I guess it is part of the requirement? This will have a bit of performance overhead, but I guess not by much since it is just metadata relation element update. Anyway, want to bring this up in case there is better way to handle this.

Reply:
The 'derived from' value is not updated once it is set when a copy of the resource is created. However, other relation types where the system sets the relation meta value as resource citation that needs to be updated when the user accessing the resource via the landing page or when downloading the resource bag as the citation may have changed since last set by the system.

Reply:
Yeah, I realized that since ```accessed on``` is now part of 'derived from' value and 'accessed on' is a ever changing value, those relation values have to be updated whenever a resource is accessed. Also, when multiple users access the same resource simultaneously, this check and save update relation meta will be called multiple times with would incur performance overhead without much gain. I seem to remember there is perhaps already a last changed by with perhaps access time as well already part of the Mezzanine page/resource model? If that is True, perhaps saving accessed on as part of the resource, and combining relation values and access time at run time when being displayed or saved? I am just thinking out load here and not sure what is the best way to handle this.

Reply:
"Yeah, I realized that since accessed on is now part of 'derived from' value and 'accessed on' is a ever changing value, those relation values have to be updated whenever a resource is accessed."

Just trying to clarify part of your comment which I have referenced here (above).  The 'derived from' value (which contains the `accessed on` date) only set by the system once when a copy of the resource is created. After that the system never updates the value stored in 'derived from'. If you look at the `update_relation_meta()` of the BaseResource class, you can see that the relation type 'source' is not part of the update.  However, we allow the user to edit this value and even delete this relation (type source) meta item.

Reply:
@hyi - the 'accessed on' date was added to the citation for the relation to indicate the date on which the relation was created (i.e., the date on which the copy was created). The point of adding the 'accessed on' date is because the resource from which the copy was made could change after the copy is made and so having the 'accessed on' date just pinpoints the time when the copy was made. So - it's set once and never changes.

Reply:
@horsburgh Thanks for the clarification which prompted me to take a quick look of the design google doc  to have a better understanding of the changes needed. This makes sense to me now.

Reply:
@pkdash thanks for the clarification. Can you clarify a bit further why ```update_relation_meta()``` needs to be called to update access time whenever a resource landing page is displayed and when the resource is downloaded for isVersionOf, isReplacedBy, isPartOf, and hasPart relation types? Can access time for those relation types be only set once similarly to this derived_from resource copy case?

Reply:
@hyi The call to `update_relation_meta()` is not for updating the access time. Since we are storing resource citation for relation type such as `isVersionOf` which is the citation of the original resource, the citation of that original resource may change if the user adds a new author to that resource or even do any update to that resource. That's why this function is called when accessing the resource landing page so that the relation type citation values displayed on the UI are not stale citations.

Reply:
@pkdash OK. That makes sense to me now. However, when a resource is replaced by a new versioned resource, this replaced resource cannot be updated any more (edit button does not show up for owners on resource landing page). So I believe at least ```isVersionOf``` relation don't have the issue of the value of the relation being changed since after a resource is being replaced by a new versioned resource, that resource cannot be updated any more meaning the citation for that resource will not change.

Reply:
@pkdash I think it is better to remove `isVersionOf` type from ````update_relation_meta()``` given the frequency this method is being called. I think the less type to check for, the better. The code is also clearer I believe.

Reply:
Removed update of `isVersionOf`